strawberryfields/engine.py

```diff
@@ -547,10 +549,11 @@ def run(self, program: Program, *, compile_options=None, **kwargs) -> Optional[R
                     return job.result
                 if job.status == "failed":
                     log.warning(
```

Comment:
More of a semantic thing, but in hindsight, it seems more appropriate to log this as an error, since the desired operation failed outright and immediate intervention is needed (e.g., run the job again or diagnose the server-side failure somehow); while I believe warnings should be primarily used to indicate if something could potentially go wrong.